onc/+onc/DataProductFile.m

```diff
@@ -96,41 +98,44 @@
                     if length(lengthData) == 1
                         this.fileSize = str2double(lengthData.Value);
                     else
-                        this.fileSize = strlength(response.Body.Data);
+                        ext = util.extractFileExtension(filename);
+                        if strcmp(ext, 'xml')
```

Comment:
You can remove a level of nesting here by:

`                   
                    ext = util.extractFileExtension(filename);
                    if length(lengthData) == 1
                        this.fileSize = str2double(lengthData.Value);
                    elseif strcmp(ext, 'xml')
                        this.fileSize = length(xmlwrite(response.Body.Data));
                    else
                        this.fileSize = strlength(response.Body.Data);
                    end
`

Reply:
Well that did not format well, one sec